packages/x-components/src/x-modules/next-queries/store/actions/fetch-next-query-preview.action.ts

```diff
@@ -0,0 +1,31 @@
+import { SearchRequest } from '@empathyco/x-types';
+import { NextQueriesXStoreModule } from '../types';
+import { XPlugin } from '../../../../plugins/index';
+
+/**
+ * Default implementation for the {@link NextQueriesActions.fetchNextQueryPreview}.
+ *
+ * @param state - The state of the store, used to retrieve the rows and the extraParams to be sent
+ * in the request.
+ * @param query - The next query to send in the request.
+ * @returns A Promise of a SearchResponse when it fetches the results, `null` if the request was
+ * not made.
+ */
+export const fetchNextQueryPreview: NextQueriesXStoreModule['actions']['fetchNextQueryPreview'] = (
+  { state },
+  query
+) => {
+  if (!query) {
+    return null;
+  }
+  const request: SearchRequest = {
+    query,
+    rows: state.config.resultsPreviewCount,
+    extraParams: state.params
```

Comment:
Ook, which should be the origin?
Something like `next_query:results`?

Reply:
in this case if not sure because the query feature is the next-query, but the feature location is the predictive_layer? Because in this case, they are not still the query inside the grid, but im not sure about this or if we should create a new feature location?